### DISCUSSION TITLE
Upgrading Postgres Database

### DISCUSSION THREAD

**Discussion by cpmills1975** [netbox-community/netbox-docker#372] [Getting Help] [answered]
I've been running a NetBox instance in development for some time via netbox-docker and have just pulled the most recent updates to bring in NetBox 2.10.0.

I notice Postgres has been updated to 12 and it won't open my DB created in Postgres 11. Can anyone offer any advice on upgrading an in-place database?

Should this be mentioned in the documentation?  In this case, I am only running a dev instance and can happily blat my DB if required, but I can't roll this out to my customers if it isn't a completely straight forward upgrade process!  I'm aware that netbox-docker is widely used for testing NetBox, but I was under the impression that it could also be used in production? If it is indeed intended for production use, the DB upgrade process needs to be simple or the DB version needs to be nailed down - will NetBox 2.10 work with Postgres 11 if required?

**Comment by cimnine** [comment]
A backup and restore of Postgres does the trick. This is the simplest way. See https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting#database-operations

> Should this be mentioned in the documentation?

I believe it was mentioned in the release notes when we upgraded from 11 to 12.

> will NetBox 2.10 work with Postgres 11 if required?

I believe so. [Netbox's docs](https://netbox.readthedocs.io/en/stable/installation/1-postgresql/) state "_NetBox requires PostgreSQL 9.6 or higher._"

**Comment by cpmills1975** [comment, marked as answer]
Thanks @cimnine. Some healthy debate on #netbox-docker, but for the purposes of helping anyone else out who stumbles across this I'm going to record that I've over-ridden the requirement for postgres:12-alpine with postgres:11-alpine (which remains supported by NetBox 2.10) by adding the following lines to my docker-compose.override.yml file:

```
postgres:
  image: postgres:11-alpine
```

It may not be the gold plated solution, but it feels like upgrading NetBox first and then upgrading PostgreSQL later (if desired) is the best way to go. Trying to do both at the same time may work, but it adds a complexity that, personally, I'd rather not have to deal with.

**Reply by cimnine** [reply]
For the interested ones: The link to the chat conversation would be https://networktocode.slack.com/archives/CD23LP8BC/p1608042926323600
